components/admin_console/reset_password_modal/reset_password_modal.tsx

```diff
@@ -1,29 +1,54 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
-import {adminResetPassword} from 'actions/admin_actions.jsx';
 import * as Utils from 'utils/utils.jsx';
 
-export default class ResetPasswordModal extends React.Component {
-    static propTypes = {
-        user: PropTypes.object,
-        currentUserId: PropTypes.string.isRequired,
-        show: PropTypes.bool.isRequired,
-        onModalSubmit: PropTypes.func,
-        onModalDismissed: PropTypes.func,
-        passwordConfig: PropTypes.object,
-    };
+interface User {
+    id: string;
+    auth_service: string;
+}
+
+interface Error {
+    message: string;
+}
+
+interface PasswordConfig {
+    minimumLength: number;
+    requireLowercase: boolean;
+    requireNumber: boolean;
+    requireSymbol: boolean;
+    requireUppercase: boolean;
+}
+
+type State = {
+    serverErrorNewPass: JSX.Element|null;
+    serverErrorCurrentPass: JSX.Element|string|null;
+}
 
-    static defaultProps = {
+type Props = {
+    user: User|null;
```

Comment:
You should be able to use the `?` shorthand to define that this object is nullable.
```suggestion
    user?: User;
```

Reply:
done